tests/test_scalars.py

```diff
@@ -15,26 +22,568 @@
+    check(assert_type(ts < c_np_ndarray_dt64, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(ts >= c_series_timestamp, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(ts < c_series_timestamp, "pd.Series[bool]"), pd.Series, bool)
+
+    check(assert_type(ts >= c_series_dt64, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(ts < c_series_dt64, "pd.Series[bool]"), pd.Series, bool)
+
+    gte = check(assert_type(c_dt_datetime >= ts, bool), bool)
+    lt = check(assert_type(c_dt_datetime < ts, bool), bool)
+    assert gte != lt
+
+    check(assert_type(c_np_dt64 >= ts, Any), np.bool_)
+    check(assert_type(c_np_dt64 < ts, Any), np.bool_)
+
+    check(assert_type(c_datetimeindex >= ts, np_ndarray_bool), np.ndarray)
+    check(assert_type(c_datetimeindex < ts, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(c_np_ndarray_dt64 >= ts, np_ndarray_bool), np.ndarray)
+    check(assert_type(c_np_ndarray_dt64 < ts, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(c_series_dt64 >= ts, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(c_series_dt64 < ts, "pd.Series[bool]"), pd.Series, bool)
+
+    eq = check(assert_type(ts == c_timestamp, bool), bool)
+    ne = check(assert_type(ts != c_timestamp, bool), bool)
+    assert eq != ne
+
+    eq = check(assert_type(ts == c_np_dt64, bool), bool)
+    ne = check(assert_type(ts != c_np_dt64, bool), bool)
+    assert eq != ne
+
+    eq = check(assert_type(ts == c_dt_datetime, bool), bool)
+    ne = check(assert_type(ts != c_dt_datetime, bool), bool)
+    assert eq != ne
+
+    check(assert_type(ts == c_datetimeindex, np_ndarray_bool), np.ndarray)
+    check(assert_type(ts != c_datetimeindex, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(ts == c_np_ndarray_dt64, np_ndarray_bool), np.ndarray)
+    check(assert_type(ts != c_np_ndarray_dt64, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(ts == c_series_timestamp, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(ts != c_series_timestamp, "pd.Series[bool]"), pd.Series, bool)
+
+    check(assert_type(ts == c_series_dt64, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(ts != c_series_dt64, "pd.Series[bool]"), pd.Series, bool)
+
+    check(assert_type(c_np_dt64 == ts, Any), np.bool_)
+    check(assert_type(c_np_dt64 != ts, Any), np.bool_)
+
+    eq = check(assert_type(c_dt_datetime == ts, bool), bool)
+    ne = check(assert_type(c_dt_datetime != ts, bool), bool)
+    assert eq != ne
+
+    check(assert_type(c_datetimeindex == ts, np_ndarray_bool), np.ndarray)
+    check(assert_type(c_datetimeindex != ts, np_ndarray_bool), np.ndarray)
+
+    check(assert_type(c_np_ndarray_dt64 != ts, Any), np.ndarray)
+    check(assert_type(c_np_ndarray_dt64 == ts, Any), np.ndarray)
+
+    check(assert_type(c_series_dt64 == ts, "pd.Series[bool]"), pd.Series, bool)
+    check(assert_type(c_series_dt64 != ts, "pd.Series[bool]"), pd.Series, bool)
```

Comment:
can remove the above 5 pairs of tests because `__eq__` and `__ne__` are always sent to the LHS object type method.

Reply:
Shouldn't we just leave them in just for good measure to make sure something doesn't break in the future?

Reply:
OK to leave in, but I think they should be in a different test function, because they are really testing the LHS of `==` and `!=` as opposed to the RHS, so the func name `test_timestamp_cmp` isn't really reflective.  

So can you split it out, and make a comment that indicates that these are tests of the LHS against `Timestamp`